exams/management/commands/populate_edx_exam_coupons.py

```diff
@@ -32,16 +33,22 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument,too-many-l
 
         csvfile = kwargs.get('csvfile')
         reader = csv.DictReader(csvfile.read().splitlines())
-        edx_course_key = next(reader)['Course ID']
+        catalog_query = next(reader)['Catalog Query']
+        course_number = re.search(r"key:\(([A-Za-z0-9.]+)", catalog_query).group(1)
 
         validated_urls = validate_urls(reader)
-        course = Course.objects.get(courserun__edx_course_key=edx_course_key)
+        course = Course.objects.filter(course_number__startswith=course_number).first()
+
+        if course is None:
```

Comment:
Can we change this to use `Course.objects.get` instead? That way there is only one course which matches.